### ISSUE TITLE
Streamline client/server gulp builds

### ISSUE THREAD

**Issue by natoverse** [atsid/permissive#80]

**Comment by bmathews** [comment]
Right now the client and server build systems are separate, and even organized differently. Ideally they'd be similar, and we might want a single entry point to start both the client and server watchers.

**Comment by Ken-2scientists** [comment]
:+1: 
This is currently making it hard for me to successfully Dockerize permissive

**Comment by bmathews** [comment]
What's the issue?

**Comment by Ken-2scientists** [comment]
In general, there's usually just one entry point in a Docker container. As far as I can tell, I can't launch both the services and serve up the UI with a single command. I could create two containers, one for the services and one to serve up the UI, but that seems like overkill here.

**Comment by bmathews** [comment]
The UI is served by the one and only main server, it just needs to be built before-hand.

**Comment by bmathews** [comment]
Do you still need one entry point to build it _and_ start the server?

**Comment by Ken-2scientists** [comment]
No, I just need to know the sequence for build, and then the right runtime.

**Comment by bmathews** [comment]
In a dream world, you'd run `gulp --release` from `/app/client` and it'd just build it for you with no watcher and then the main server could serve it up, but that's throwing an error since we've never tried it. :) Working on that now.

**Comment by bmathews** [comment]
`gulp --release` works now.

so something like this maybe `npm install && gulp && cd /app/client && gulp --release && cd ../../ && npm start`

**Comment by Ken-2scientists** [comment]
I'll give it a shot. I think that `npm install && gulp && cd /app/client && gulp --release` should probably do it for the Dockerfile build and then I can leave `npm start` as the default command for the container.
